### DISCUSSION TITLE
Orientation inconsistency in plotting crystalShape in Polefigure

### DISCUSSION THREAD

**Discussion by Tijmenvermeij** [mtex-toolbox/mtex#1682] [Ask Anything]
Hi,

I've recently been checking if my EBSD data has been correctly imported into MTex, through comparison of plotted crystalShapes and also through comparison of polefigure distributions. I found that the crystalshapes match the EBSD software (EBSD), but the polefigures do not. 

I've looked a bit into it and have a theory of what the problem could be. I think that the 'upper' part of a polefigure plot is not actually the upper hemisphere w.r.t. the viewing direction (i.e. out of the screen), but instead in the direction of the positive Z-plotting direction. I always use zIntoPlane for Mtex plotting...

Here is a bit of code to show this problem. I define a random HCP orientation and plot the crystalShape into a polefigure, at the c-axis (0001) location, for zOutOfPlane and zIntoPlane:

```
%% test plotting of a shape

% define CS
CS = crystalSymmetry('6/mmm');

% define random orientation
ori = orientation.rand(CS);

% define hex shape
cS = crystalShape.hex(CS);

% define c-axis
c = Miller(0,0,1,CS);

% try plotting with z into-plane
plotx2east;
plotzIntoPlane;

figure;
plotPDF(ori,c,'complete','grid');
plot(ori,ori*cS,'add2all');

% try plotting with z out-of-plane
plotx2east;
plotzOutOfPlane;

figure;
plotPDF(ori,c,'complete','grid');
plot(ori,ori*cS,'add2all');
```

This gives the following images:
![ZIntoPlane](https://user-images.githubusercontent.com/16989758/205930813-9581246f-d855-46b6-ba11-406af8f571cb.png)
![ZOutOfPlane](https://user-images.githubusercontent.com/16989758/205930848-b5cf961c-035d-4331-9521-fa39aef26bf6.png)

You can clearly seen that the zOutOfPlane plot is correctly plotting the hexagonal shape in the polefigure, with the c-axis pointing in the direction according to the position in which it is plotted, while the zIntoPlane plot is not.

Any idea what's going on here? Is this on purpose? If so, I don't find it very logical...

**Comment by ralfHielscher** [comment]
Hi Tijmen,

You are completely right that the confusion comes from the question what is the "upper" hemisphere. You can define this either 
 1. by plotting convention, i.e., which hemisphere sticks out of the plotting surface, or
 2. by the object which is being plotted, i.e., the hemisphere with with positive z values. (This is what MTEX does) 

The reasoning we went in MTEX with definition (2) was that we would like to do
```matlab
v = equispacedS2Grid('upper')
```
to define a grid on the northern hemisphere, no matter the current plotting convention. 
However, I see your point and since we are currently planning to in introduce 
 *  pseudo 3d EBSD 
 * different specimen reference frames
 * arbitrary plotting alignment of these reference frames
it might be a good idea to rethink this convention.  One could indeed say that
```matlab
v = equispacedS2Grid('upper')
```
should always refer to the current plotting convention while
```matlab
v = equispacedS2Grid('northern')
```
could be the plotting independent definition.

What do you think?
All the best,
Ralf.

**Reply by Tijmenvermeij** [reply]
Hi Ralf,

Thanks for the explanation.

I understand the reasoning behind the definition of upper. I guess most people use the default Mtex settings (zOutofPlane) anyways, so they don't suffer from this.

I like the definition of 'northern', but I'm still a bit worried with people using zIntoPlane who are completely unaware of this problem. You might say they should check their polefigures and compare with commercial system, but I think it makes sense to either (i) give a warning when 'upper' is not actually pointing up (north) or (ii) use 'northern' as a default.

Best,
Tijmen
